### ISSUE TITLE
Whoops! /run/nordvpn/nordvpnd.sock not found

### ISSUE THREAD

**Issue by mlukewizard** [bubuntux/nordvpn#282]
Error when using the docker compose example in the readme. Run with `docker-compose up`. docker-compose.yml looks like:

```
version: "3"
services:
  vpn:
    image: ghcr.io/bubuntux/nordvpn
    cap_add:
      - NET_ADMIN               # Required
    environment:                # Review https://github.com/bubuntux/nordvpn#environment-variables
      - USER=user@email.com     # Required
      - "PASS=pas$word"         # Required
      - CONNECT=United_States
      - TECHNOLOGY=NordLynx
      - NETWORK=192.168.1.0/24  # So it can be accessed within the local network
    ports:
      - 8080:8080
    sysctls:
      - net.ipv6.conf.all.disable_ipv6=1  # Recomended if using ipv4 only
  torrent:
    image: ghcr.io/linuxserver/qbittorrent
    network_mode: service:vpn
    depends_on:
      - vpn
```
      

I realise the error message says 'Invalid Username or password', but I can guarantee I have typed them in correctly. The error is in the line: "Whoops! /run/nordvpn/nordvpnd.sock not found". The same error is produced when running nordvpn status.

Any ideas why the container errors and I can't connect to nord?

Trace is:

```
Recreating qbitorrent-docker_vpn_1 ... done
Recreating qbitorrent-docker_torrent_1 ... done
Attaching to qbitorrent-docker_vpn_1, qbitorrent-docker_torrent_1
vpn_1      | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
vpn_1      | [s6-init] ensuring user provided files have correct perms...exited 0.
vpn_1      | [fix-attrs.d] applying ownership & permissions fixes...
vpn_1      | [fix-attrs.d] docker_utils: applying...
vpn_1      | [fix-attrs.d] docker_utils: exited 0.
torrent_1  | [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
vpn_1      | [fix-attrs.d] nord_utils: applying...
vpn_1      | [fix-attrs.d] nord_utils: exited 0.
vpn_1      | [fix-attrs.d] done.
vpn_1      | [cont-init.d] executing container initialization scripts...
vpn_1      | [cont-init.d] 00-firewall: executing...
vpn_1      | Firewall is up, everything has to go through the vpn
vpn_1      | [cont-init.d] 00-firewall: exited 0.
vpn_1      | [cont-init.d] 10-tun: executing...
vpn_1      | [cont-init.d] 10-tun: exited 0.
vpn_1      | [cont-init.d] 20-inet: executing...
vpn_1      | Enabling connection to secure interface and docker network
torrent_1  | [s6-init] ensuring user provided files have correct perms...exited 0.
torrent_1  | [fix-attrs.d] applying ownership & permissions fixes...
torrent_1  | [fix-attrs.d] done.
torrent_1  | [cont-init.d] executing container initialization scripts...
torrent_1  | [cont-init.d] 01-envfile: executing...
torrent_1  | [cont-init.d] 01-envfile: exited 0.
torrent_1  | [cont-init.d] 10-adduser: executing...
torrent_1  | usermod: no changes
torrent_1  |
torrent_1  | -------------------------------------
torrent_1  |           _         ()
torrent_1  |          | |  ___   _    __
torrent_1  |          | | / __| | |  /  \
torrent_1  |          | | \__ \ | | | () |
torrent_1  |          |_| |___/ |_|  \__/
torrent_1  |
torrent_1  |
torrent_1  | Brought to you by linuxserver.io
torrent_1  | -------------------------------------
torrent_1  |
torrent_1  | To support LSIO projects visit:
torrent_1  | https://www.linuxserver.io/donate/
torrent_1  | -------------------------------------
torrent_1  | GID/UID
torrent_1  | -------------------------------------
torrent_1  |
torrent_1  | User uid:    911
torrent_1  | User gid:    911
torrent_1  | -------------------------------------
torrent_1  |
vpn_1      | [cont-init.d] 20-inet: exited 0.
vpn_1      | [cont-init.d] 20-inet6: executing...
torrent_1  | [cont-init.d] 10-adduser: exited 0.
vpn_1      | No inet6 network
vpn_1      | [cont-init.d] 20-inet6: exited 0.
torrent_1  | [cont-init.d] 30-config: executing...
vpn_1      | [cont-init.d] 30-route: executing...
torrent_1  | [cont-init.d] 30-config: exited 0.
vpn_1      | Enabling connection to network 192.168.1.0/24
torrent_1  | [cont-init.d] 90-custom-folders: executing...
vpn_1      | [cont-init.d] 30-route: exited 0.
torrent_1  | [cont-init.d] 90-custom-folders: exited 0.
vpn_1      | [cont-init.d] 30-route6: executing...
torrent_1  | [cont-init.d] 99-custom-scripts: executing...
vpn_1      | [cont-init.d] 30-route6: exited 0.
vpn_1      | [cont-init.d] 40-allowlist: executing...
vpn_1      | [cont-init.d] 40-allowlist: exited 0.
vpn_1      | [cont-init.d] done.
vpn_1      | [services.d] starting services
torrent_1  | [custom-init] no custom files found exiting...
torrent_1  | [cont-init.d] 99-custom-scripts: exited 0.
torrent_1  | [cont-init.d] done.
torrent_1  | [services.d] starting services
vpn_1      | 2021/11/14 23:58:50 [Info] Daemon has started
torrent_1  | [services.d] done.
vpn_1      | 2021/11/14 23:58:50 Get "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:50 error on making client request: Head "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:50 cdn api: X-Accept-Before UNIX value is lower than current local time
vpn_1      | 2021/11/14 23:58:51 [Info] Daemon has started
vpn_1      | 2021/11/14 23:58:51 Get "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:51 error on making client request: Head "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:51 cdn api: X-Accept-Before UNIX value is lower than current local time
vpn_1      | 2021/11/14 23:58:52 [Info] Daemon has started
vpn_1      | 2021/11/14 23:58:52 Get "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:52 error on making client request: Head "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:52 cdn api: X-Accept-Before UNIX value is lower than current local time
vpn_1      | 2021/11/14 23:58:53 [Info] Daemon has started
vpn_1      | 2021/11/14 23:58:53 Get "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:53 error on making client request: Head "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:53 cdn api: X-Accept-Before UNIX value is lower than current local time
vpn_1      | 2021/11/14 23:58:54 [Info] Daemon has started
vpn_1      | 2021/11/14 23:58:54 Get "https://cdn.zwyr157wwiu6eior.com/configs/dns/cybersec.json": EOF
vpn_1      | 2021/11/14 23:58:54 cdn api: X-Accept-Before UNIX value is lower than current local time
vpn_1      | s6-svwait: fatal: timed out
Whoops! /run/nordvpn/nordvpnd.sock not found
vpn_1      | Invalid Username or password.
```

**Comment by mlukewizard** [comment]
Worth adding: I see the same error when launching the NordVPN docker container by itself



```
docker run -ti --cap-add=NET_ADMIN --name vpn \
           -e USER=user@email.com -e PASS='pas$word' \
           -e TECHNOLOGY=NordLynx -d ghcr.io/bubuntux/nordvpn
```

**Comment by bubuntux** [comment]
i'm not able to reproduce the issue, in any case y would recommend you use https://github.com/bubuntux/nordlynx instead

**Comment by MrLinford** [comment]
I just switched to https://github.com/bubuntux/nordlynx after I kept getting an Invalid login error and it works. I will monitor though.

**Comment by alvinuk** [comment]
I too am also getting the same error as @mlukewizard

**Comment by HenryDanks00** [comment]
Same here.

**Comment by mlukewizard** [comment]
I've had it working at various times tonight, and not working at others. Same file, no changes, works sometimes, doesn't work other times... can't figure out what's going on.

**Comment by bubuntux** [comment]
hmmm that is odd, wondering if the issue is on s6

**Comment by giyeol** [comment]
same here

**Comment by mlukewizard** [comment]
Was working fine for me yesterday

**Comment by tchrischan** [comment]
I have had to restart nordvpn every few hours; it works for a little while (long enough for me to confirm it's working) then I get the "whoops" error. I was not seeing the invalid password error though; I was seeing some messages about not being able to contact the gateway.

Anyway, I had already switched to v3.10.0-1-1 to see if that fixes it, though I don't think it's been long enough to see if it will stay up.

**Comment by MRobi1** [comment]
I've got a slightly different issue but it was linked back to this one as the same.

If I run v3.10.0-1-1 everything connects on deployment. 
I'm getting "A new version of NordVPN is available! Please update the application." in the logs but it connects just fine.
The issue I'm having with v3.10.0-1-1 is that anytime I stop the container (ie: a restart), it will no longer connect. It just tries and stops over and over. I have to remove and redeploy. Not the end of the world but a pain nonetheless.

Any version I run beyond v3.10.0-1-1 does not launch at all. 
Log just shows "/bin/sh: 1: /usr/bin/start_vpn.sh: not found" over and over and over again. Nothing else.
This includes:
v3.12.0-1-1
v3.12.0-1-2
v3.12.1.1-1
master
latest

All give the exact same log message. The only difference in the config is the version being pulled.

I tried playing with the NordLynx container, can get it connecting to NordVPN. Can connect another container through it. But haven't been able to access any containers I set to run through it. From the documentation it doesn't seem to have the NETWORK env variable and the handling of ports seems to be different as well.

**Comment by tchrischan** [comment]
Should report back v3.10.0-1-1 has been up for 6 days.

**Comment by MRobi1** [comment]
> Should report back v3.10.0-1-1 has been up for 6 days.

Yes. For me v3.10.0-1-1 will stay up and connected for however long I leave it running. But if I stop the container for what seems like any reason (usually a system reboot), it will not reconnect for me. I have to remove and re-deploy the container to get a connection again. It's odd.

**Comment by mlukewizard** [comment]
Just attempted to reboot my VPN container, seeing this issue again...

**Comment by mlukewizard** [comment]
The v3.10.0-1-1 tag works nicely for me

**Comment by cschinis** [comment]
Same issue here. It works fine for a few hours and then it gives me the Whoops! /run/nordvpn/nordvpnd.sock not found error. v3.10.0-1-1 unfortunately doesn't work for me.

**Comment by bubuntux** [comment]
**TL;DR**  just use migrate to https://github.com/bubuntux/nordlynx or fallback to v3.10.0-1-1

**Comment by kolonuk** [comment]
In case anyone else comes here with this issue, either 
1. Wait 10 minutes and try again.  It seems that NordVPN returns invalid username/password for too many re-connects in a short time.
2. Also, make sure that your username and password are wrapped in double quotes.
